From 54f7fa6ece0c455b49495ccca3cc412a583ccb43 Mon Sep 17 00:00:00 2001
From: Sora Lee <saih792@outlook.com>
Date: Mon, 14 May 2018 00:07:53 +1200
Subject: [PATCH] Add support for libressl.

---
 buckets/ssl_buckets.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/buckets/ssl_buckets.c b/buckets/ssl_buckets.c
index b01e535..eeae556 100644
--- buckets/ssl_buckets.c
+++ buckets/ssl_buckets.c
@@ -52,7 +52,7 @@
 #define APR_ARRAY_PUSH(ary,type) (*((type *)apr_array_push(ary)))
 #endif
 
-#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L
+#if defined(OPENSSL_VERSION_NUMBER) && OPENSSL_VERSION_NUMBER >= 0x10100000L && !defined(LIBRESSL_VERSION_NUMBER)
 #define USE_OPENSSL_1_1_API
 #endif
 
-- 
2.16.1

